### PR TITLE
Adds global CTA banner 

### PR DESCRIPTION
**Pull request by artemruts**
### Description

We need to show two CTA banners to the users that have a GitHub code host connection but it's missing the `repo` scope. This prevents users to access their private repositories.

These changes add a global dismissible banner for user that is:
* on `.com`
* enrolled in private code
* is missing "private" scope for GitHub

React `context` was used here since I had "communicate" with two banners that are in a very different places in the render tree. I felt that passing props would create more problems and make the code less maintainable.

___

**Note:** this is a temporarily solution. CTA banners will be removed once the Core app team confirms that all of the users with GitHub code host connection have the `repo` scope.
___

### Demo

Remove

https://user-images.githubusercontent.com/1319181/121422845-9e3e4080-c93d-11eb-8835-4eeddf21ebff.mov

Update

https://user-images.githubusercontent.com/1319181/121422874-a6967b80-c93d-11eb-861a-a83c3ccb3dd8.mov

Dissmiss

https://user-images.githubusercontent.com/1319181/121422916-b44c0100-c93d-11eb-8ac6-4dacc76de098.mov

### CODE REVIEW

**Review by quinnkeast — APPROVED** [review]
Looks great!

**Review by unknwon — APPROVED** [review]
Looking great! (didn't review code)

**Review by ryanslade — APPROVED** [review]
Nice!

File: client/web/src/user/settings/codeHosts/UserAddCodeHostsPage.tsx [hunk 1]
```diff
@@ -64,7 +65,16 @@ export const UserAddCodeHostsPage: React.FunctionComponent<UserAddCodeHostsPageP
     onUserExternalServicesOrRepositoriesUpdate,
 }) => {
     const [statusOrError, setStatusOrError] = useState<Status>()
-    const [updateAuthRequired, setUpdateAuthRequired] = useState(false)
+    // const [isGitHubTokenUpdateRequired, setIsGitHubTokenUpdateRequired] = useState(false)
```
Comment (tjkandala) [inline client/web/src/user/settings/codeHosts/UserAddCodeHostsPage.tsx:68]:
```suggestion
```

**Review by tjkandala — APPROVED** [review]
LGTM

**Pull request merged**